packages/kilt-did/README.md

```diff
@@ -3,6 +3,320 @@
+
+An example of a full DID is the following:
+
+```
+did:kilt:4rp4rcDHP71YrBNvDhcH5iRoM3YzVoQVnCZvQPwPom9bjo2e
+```
+
+Here, there is no `light:` component, which indicates that the DID is a full DID and that the keys associated with it must not be derived from the DID identifier but must be retrieved from the KILT blockchain.
+
+Beyond an authentication key, and encryption key, and external services, a full DID also supports an **attestation key**, which must be used to write CTypes and attestations on the blockchain, and a **delegation key**, which must be used to write delegations on the blockchain.
+
+### Creating and anchoring a full DID
+
+The following is an example of how to create and write a full DID on chain that specifies only an authentication key.
+
+```typescript
+import Keyring from '@polkadot/keyring'
+
+import Kilt from '@kiltprotocol/sdk-js'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+import {
+  DefaultResolver,
+  DemoKeystore,
+  DidUtils,
+  SigningAlgorithms
+} from '@kiltprotocol/did'
+import { KeyRelationship } from '@kiltprotocol/types'
+
+// Initialise connection to a KILT blockchain node.
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
+
+// Generate the KILT account that will submit the DID creation tx to the KILT blockchain.
+// It must have enough funds to pay for the tx execution fees.
+const aliceKiltAccount = new Keyring({
+  type: 'ed25519',
+  // KILT has registered the ss58 prefix 38
+  ss58Format: 38,
+}).createFromUri('//Alice')
+
+// Instantiate the demo keystore.
+const keystore = new DemoKeystore()
+
+// Generate seed for the authentication key.
+const authenticationSeed = '0x123456789'
+
+// Ask the keystore to generate a new keypar to use for authentication.
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate the DID-signed creation extrinsic.
+// The extrinsic is unsigned and contains the DID creation operation signed with the DID authentication key.
+const { submittable, did } = await DidUtils.writeDidfromPublicKeys(keystore, {
+  [KeyRelationship.authentication]: {
+    publicKey: authenticationKeyPublicDetails.publicKey,
+    type: authenticationKeyPublicDetails.alg,
+  },
+})
+// Will print `did:kilt:4sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar`.
+console.log(did)
+
+// Submit the DID creation tx to the KILT blockchain after signing it with the KILT account specified.
+await BlockchainUtils.signAndSubmitTx(submittable, aliceKiltAccount, {
+  resolveOn: BlockchainUtils.IS_IN_BLOCK,
+})
+
+const fullDid = (await DefaultResolver.resolveDoc(did)) as FullDidDetails
```

Comment:
First, `FullDidDetails` is not imported in this example. However, my suggestion is to remove `as FullDidDetails` from here completely. I assume the reason for it was that resolveDoc returns `FullDidDetails | null`, and I want to challenge that as well. Do you think that "DID not found" is an error deserving a throw?

Reply:
Thanks for pointing this out. Since `fullDid` is anyway not used anywhere else, it makes sense to remove `as FullDidDetails`. It is up to the user to decide what to do with a potentially `null` value.

Reply:
Do you think that `resolveDoc` should throw an error instead of returning null?

Reply:
`resolveDoc` does return `Promise<IDidDetails | null>` already. And I think that is the correct way. Only in the example I was casting it because we are sure the DID exists as it was created with no issues. But I fixed that.